### PR TITLE
E UI 4824 search location seperation another user scenario

### PR DESCRIPTION
**Pull request by waynepHMCTS**
Resolves # . (This is applicable only if this pull request relates to a GitHub issue, delete the line otherwise)

Notes:
* This is to refactor the search location and extend the serviceIds to at mulitple search capabilities
*
*

### CODE REVIEW

**Review by johnbenjamin-hmcts — APPROVED** [review]

File: projects/exui-common-lib/src/lib/components/search-location/search-location.component.ts [hunk 1]
```diff
@@ -12,24 +12,25 @@ import {LocationService} from '../../services/locations/location.service';
...
   public selectedLocation: LocationByEPIMSModel;
-  public showAutocomplete = false;
 
   private readonly minSearchCharacters = 3;
+  public keyUpSubject$: Subject<string> = new Subject();
 
   constructor(private readonly locationService: LocationService, fb: FormBuilder) {
     this.findLocationForm = fb.group({
       findLocationControl: [null],
+      locationSelected: [null]
```
Comment (rayliangatos) [inline projects/exui-common-lib/src/lib/components/search-location/search-location.component.ts:33]:
```suggestion
      findLocationFormControl: [null],
      locationSelectedFormControl: [null]
```

File: projects/exui-common-lib/src/lib/components/search-location/search-location.component.ts [hunk 2]
```diff
@@ -38,58 +39,63 @@ export class SearchLocationComponent implements OnInit {
   public ngOnInit(): void {
     this.locations$ = this.getLocations('');
```
Comment (rayliangatos) [inline projects/exui-common-lib/src/lib/components/search-location/search-location.component.ts:40]:
```suggestion
    this.locations$ = of([]);
```

File: projects/exui-common-lib/src/lib/components/search-location/search-location.component.ts [hunk 3]
```diff
@@ -38,58 +39,63 @@ export class SearchLocationComponent implements OnInit {
...
+    if (this.control) {
+      if (this.findLocationForm && this.findLocationForm.controls) {
+        this.findLocationForm.controls.locationSelected =  this.control;
+      }
+    }
+
+    this.keyUpSubject$.pipe(debounceTime(500)).subscribe(searchValue => this.search(searchValue as string));
+  }
+
+  public onKeyUp(event: any) {
```
Comment (rayliangatos) [inline projects/exui-common-lib/src/lib/components/search-location/search-location.component.ts:51]: return type

File: projects/exui-common-lib/src/lib/components/search-location/search-location.component.ts [hunk 4]
```diff
@@ -38,58 +39,63 @@ export class SearchLocationComponent implements OnInit {
...
+      mergeMap((locations: LocationByEPIMSModel[]) => this.selectedLocations$.pipe(
+          map((selectedLocations) => locations.filter(
+            location => !selectedLocations.map(selectedLocation => selectedLocation.epims_id).includes(location.epims_id)
+          )),
+        )
+      )
+    ) : of([]);
   }
 
   public filter(term: string) {
```
Comment (rayliangatos) [inline projects/exui-common-lib/src/lib/components/search-location/search-location.component.ts:66]: return type

File: projects/exui-common-lib/src/lib/components/search-location/search-location.component.ts [hunk 5]
```diff
@@ -38,58 +39,63 @@ export class SearchLocationComponent implements OnInit {
...
     ).subscribe(locations => {
       this.locations$ = of(locations);
     });
   }
 
+  public onSelectionChange(selection?: LocationByEPIMSModel) {
+    this.findLocationForm.controls.locationSelected.setValue(selection);
+  }
+
   public search(currentValue: string) {
```
Comment (rayliangatos) [inline projects/exui-common-lib/src/lib/components/search-location/search-location.component.ts:82]: return type

File: projects/exui-common-lib/src/lib/components/search-location/search-location.component.ts [hunk 6]
```diff
@@ -38,58 +39,63 @@ export class SearchLocationComponent implements OnInit {
...
-  public removeSelection(location: LocationByEPIMSModel) {
-    this.selectedLocations$.subscribe(selectedLocations => {
-      const index = selectedLocations.findIndex(selectedLocation => selectedLocation.epims_id === location.epims_id);
-      selectedLocations.splice(index, 1);
-    });
-  }
-
-  public getDisplayName(selectedLocation: LocationByEPIMSModel): string {
-    return selectedLocation.court_name;
+  public getControlCourtNameValue() {
```
Comment (rayliangatos) [inline projects/exui-common-lib/src/lib/components/search-location/search-location.component.ts:97]: return type

File: projects/exui-common-lib/src/lib/components/search-location/search-location.component.ts (hunk 3 above)
Comment (waynepHMCTS) [reply projects/exui-common-lib/src/lib/components/search-location/search-location.component.ts:51]: return type for what @rayliangatos

Comment (waynepHMCTS) [reply projects/exui-common-lib/src/lib/components/search-location/search-location.component.ts:51]: what line

File: projects/exui-common-lib/src/lib/components/search-location/search-location.component.ts [hunk 7]
```diff
@@ -12,24 +12,25 @@ import {LocationService} from '../../services/locations/location.service';
 })
 
 export class SearchLocationComponent implements OnInit {
-  // i.e. serviceIds = SSCS or serviceIds = SSCS,IA,EMPLOYMENT etc.
   @Input() public serviceIds: string = '';
-  // i.e. locationType = optional or hearing or case_management
   @Input() public locationType: string = '';
-  @Input() public disabled?: boolean = null;
+  @Input() public disabled: boolean = false;
```
[the remark this answers is not in the document]
Comment (rayliangatos) [reply projects/exui-common-lib/src/lib/components/search-location/search-location.component.ts]: is the optional still needed?

File: projects/exui-common-lib/src/lib/components/search-location/search-location.component.html [hunk 8]
```diff
@@ -1,44 +1,24 @@
...
+    <input                             
+      [formControl]="findLocationFormGroup.controls.findLocationFormControl"
+      [matAutocomplete]="auto"
+      (keyup)="onKeyUp($event)"
+      aria-expanded="false" 
+      aria-owns="input-selected-location__listbox" 
+      aria-autocomplete="both" 
+      autocomplete="off" 
+      class="autocomplete__input autocomplete__input--default search-box"
+      [attr.disabled]="!disabled ? null: disabled"
```
Comment (rayliangatos) [inline projects/exui-common-lib/src/lib/components/search-location/search-location.component.html:12]: is this needed to be reverted?

Comment (waynepHMCTS) [reply projects/exui-common-lib/src/lib/components/search-location/search-location.component.html:12]: Yes in case the user off the property puts false in

**Review by rayliangatos — APPROVED** [review]
thanks addressing the issues, approved

File: package.json [hunk 9]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@hmcts/rpx-xui-common-lib",
-  "version": "1.4.13-location-api-cr",
+  "version": "tag 1.4.19-search-location-fixes",
```
Comment (rayliangatos) [inline package.json:3]: what is this tag before version

**Pull request merged**